### PR TITLE
Add FBGEMM_ENUM_CLASS_API

### PR DESCRIPTION
**Pull request by shz0116**
On macos, enum class needs to be declared visibility attributes explicitly for shared build, but not on ubuntu.

### CODE REVIEW

File: include/fbgemm/FbgemmBuild.h [hunk 1]
```diff
@@ -25,14 +26,19 @@
...
-    // "GNUC with version < 6 have a bug if visibility attribute is used on an enum class.
-    // https://stackoverflow.com/questions/2463113/g-c0x-enum-class-compiler-warnings "
-    #if __clang__ || __GNUC__ >=6 || __INTEL_COMPILER
+    #if __clang__ || __GNUC__ >= 4 || __INTEL_COMPILER
       #define FBGEMM_API __attribute__((__visibility__("default")))
     #else
       #define FBGEMM_API
     #endif
+    // Currently, enum classes need to be declaredly explicitly for shared build on macos
+    #if __clang__
```
Comment (dskhudia) [inline include/fbgemm/FbgemmBuild.h]: clang builds are used on linux as well.

**Comment by dskhudia** [comment]
Please make a PyTorch OSS test PR.

File: include/fbgemm/FbgemmBuild.h (hunk 1 above)
Comment (shz0116) [reply include/fbgemm/FbgemmBuild.h]: Need to find whether this is related with __clang__ or __APPLE__.

**Comment by shz0116** [comment]
> Please make a PyTorch OSS test PR.

All tests are passed for PyTorch OSS.  See https://github.com/pytorch/pytorch/pull/32191

**Pull request closed without merging**